lib/rust/profiler/data/src/aggregate.rs

```diff
@@ -0,0 +1,87 @@
+//! Supports aggregating interval data by profiler to analyze total time spent, abstracting away
+//! *when* intervals occurred.
+
+use std::collections;
+use std::rc;
+
+/// Aggregate time spent in different functions.
+#[derive(Default, Debug)]
+pub struct Aggregator {
+    stack: Vec<std::rc::Rc<String>>,
+    root:  Frame,
+}
+
```

Comment:
no sections. Please remember that each file should contain at least one section.